File: frame/v2.0/replicad/motor_cap.js
const { makeBox, makeCylinder } = replicad;

const unitSize = 8.0;           // lego unit size
const unitHeight = 9.6;         // lego brick height
const holeInnerDiameter = 4.8;  // lego hole inner diameter
const holeOuterDiameter = 6.2;  // lego hole outer diameter

const main = () => {
    // Base
    const length = unitSize * 7;
    const width = unitSize * 12;
    const height = unitHeight / 3;
    
    let part = makeBox(
        [-length/2, -width/2, 0],
        [length/2, width/2, height]
    );
    
    // Cutout
    const cutoutLength = unitSize * 4;
    const cutoutWidth = unitSize * 10;
    const cutoutHeight = unitHeight / 3;
    
    let cutout = makeBox(
        [-cutoutLength/2, -cutoutWidth/2, 0],
        [cutoutLength/2, cutoutWidth/2, cutoutHeight]
    ).translate(unitSize * 0.5, 0, 0);
    part = part.cut(cutout);
    
    // Mounting screw holes
    const screwRadius = 1.5;
    const screwHeight = unitHeight;
    
    const screwPositions = [
        [unitSize * -2.5, unitSize * 5.5, 0],
        [unitSize * -2.5, unitSize * -5.5, 0],
        [unitSize * 3, unitSize * 4, 0],
        [unitSize * 3, unitSize * -4, 0],
    ];
    
    for (const pos of screwPositions) {
        let screwHole = makeCylinder(screwRadius, screwHeight, [0, 0, 0], [0, 0, screwHeight])
            .translate(pos[0], pos[1], pos[2]);
        part = part.cut(screwHole);
    }
    
    // Apply fillet to all edges that are parallel to Z-axis
    part = part.fillet(2, (e) => e.inDirection('Z'));
    
    return part;
}

File: frame/v2.0/build123d/motor_cap.py

from build123d import *
from ocp_vscode import show

unit_size           = 8.0 # lego unit size
unit_height         = 9.6 # lego brick height
hole_inner_diameter = 4.8 # lego hole inner diameter
hole_outer_diameter = 6.2 # lego hole outer diameter

with BuildPart() as part:
    # База
    length = unit_size * 7
    width  = unit_size * 12
    height = unit_height / 3
    Box(length, width, height)
    
    # Вырезы
    x_pos = unit_size * 0.5
    y_pos = 0
    z_pos = 0
    with BuildPart(Location((x_pos, y_pos, z_pos)), mode=Mode.SUBTRACT):
        length = unit_size * 4
        width  = unit_size * 10
        height = unit_height / 3
        align  = (Align.CENTER, Align.CENTER, Align.CENTER)
        Box(length, width, height)
    
    # Отверстия под крепежные винты
    points = [
        (unit_size * -2.5, unit_size * 5.5,  0),
        (unit_size * -2.5, unit_size * -5.5, 0),
        (unit_size * 3,  unit_size * 4,  0),
        (unit_size * 3,  unit_size * -4, 0),
    ]
    for point in points:
        with BuildPart(Location(point), mode=Mode.SUBTRACT):
            radius = 1.5
            height = unit_height
            align  = (Align.CENTER, Align.CENTER, Align.CENTER)
            Cylinder(radius, height, align=align)

    roundable_edges = part.edges().filter_by(Axis.Z)
    fillet(roundable_edges, radius=2)

filename = f'{__file__[:-3]}'

show(part.part, names=[filename])
export_stl(part.part, f'{filename}.stl')
